lam4-backend/src/Lam4/Expr/ConcreteSyntax.hs

```diff
@@ -58,16 +69,18 @@ TODO:
 data Expr
   = Var        Name
   | Lit        Lit
+  | Cons       Expr Expr                           -- list cons
```

Comment:
What's the difference between this and `List [Expr]`?

Reply:
Good question! These two constructs were taken from the Simala AST, but basically `List [Expr]` is meant to be the construct capable of representing an empty list. `Cons Expr Expr` wouldn't be enough if we didn't also have a way to represent the empty list.